lib/modules/schedule/page/schedule_home_page.dart

```diff
@@ -0,0 +1,164 @@
+import '../widget/home_widgets/clocking_widget.dart';
+import '../widget/home_widgets/header_widget.dart';
+
+class ScheduleHomePage extends StatefulWidget {
+  const ScheduleHomePage({Key? key}) : super(key: key);
+
+  @override
+  State<ScheduleHomePage> createState() => _SchedulePageState();
+}
+
+class _SchedulePageState extends State<ScheduleHomePage> {
+  ScheduleItemEntity getToday(List<ScheduleItemEntity> schedules) =>
+      schedules.firstWhere(
+          (element) => (element.dateTime.year == DateTime.now().year &&
+              element.dateTime.month == DateTime.now().month &&
+              element.dateTime.day == DateTime.now().day),
+          orElse: () => ScheduleItemEntity(
+              status: "Normal Shift",
+              shiftStart: "9:00 AM",
+              shiftEnd: "6:00 PM",
+              dateTime: DateTime.now()));
+
+  List<HomeCardEntity> menu = [
+    HomeCardEntity(menuIcon: Icons.wallet_membership, title: "Payslip"),
+    HomeCardEntity(menuIcon: Icons.book, title: "Directory"),
+    HomeCardEntity(menuIcon: Icons.bolt_outlined, title: "Instacash"),
+    HomeCardEntity(menuIcon: Icons.calendar_today, title: "Attendance"),
+    HomeCardEntity(menuIcon: Icons.payment, title: "Payments"),
+    HomeCardEntity(menuIcon: Icons.web, title: "Insight"),
+    HomeCardEntity(menuIcon: Icons.image, title: "Label"),
+    HomeCardEntity(menuIcon: Icons.person_pin, title: "201"),
+    HomeCardEntity(menuIcon: Icons.fingerprint, title: "Biologs"),
+    HomeCardEntity(menuIcon: Icons.groups, title: "Recruit"),
+    HomeCardEntity(menuIcon: Icons.favorite, title: "HMO"),
+    HomeCardEntity(menuIcon: Icons.more_horiz, title: "More"),
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+          centerTitle: true,
+          title: Text(
+            "Company name",
+            style: TextStyle(
+              fontWeight: FontWeight.w500,
+              fontSize: 17,
+              color: CommonColors.neutral[800],
+            ),
+          ),
+          backgroundColor: CommonColors.transparent,
+          elevation: 0,
+          bottom: PreferredSize(
+            child: Container(
+              color: CommonColors.neutral[30],
+              height: 1.0,
+            ),
+            preferredSize: Size.fromHeight(1),
+          ),
+          leading: Builder(builder: (context) {
+            return IconButton(
+              onPressed: () {
+                Scaffold.of(context).openDrawer();
+              },
+              icon: Icon(
+                Icons.menu,
+                color: CommonColors.green[300],
+              ),
+            );
+          })),
+      drawer: AppDrawer(),
+      body: BlocProvider(create: (BuildContext context) {
+        return ScheduleBloc();
+      }, child: BlocBuilder<ScheduleBloc, ScheduleState>(
+        builder: (context, state) {
+          return state.when(initial: () {
```

Comment:
remove mapping of `initial` state. Move this to initState to fully use the `StatefulWidget`